src/discrete.jl

```diff
@@ -56,7 +84,18 @@ function uniform_belief(pomdp)
     return DiscreteBelief(pomdp, state_list, ones(ns) / ns)
 end
 
-pdf(b::DiscreteBelief, s) = b.b[stateindex(b.pomdp, s)]
+# TODO suggestion: If we find that uniform_belief is necessarry for staged DiscreteBeliefs,
+# we can add uniform_belief(pomdp) which creates DiscreteBelief for first stage
+# or define new uniform_bleief(pomdp, stage) for a given stage
+# Something like:
+# uniform_belief(::FiniteHorizon, pomdp, stage) {...}
+# uniform_belief(::FiniteHorizon, pomdp) = uniform_belief(::FiniteHorizon, pomdp, 1)
+# uniform_belief(pomdp) = uniform_belief(HorizonLength(pomdp), pomdp, 1)
+# uniform_belief(pomdp, stage) = uniform_belief(HorizonLength(pomdp), pomdp, stage)
+
+pdf(b::DiscreteBelief, s) = pdf(HorizonLength(b.pomdp), b, s)
+pdf(::FiniteHorizon, b::DiscreteBelief, s) = b.b[stage_stateindex(b.pomdp, s)]
```

Comment:
This won't quite work correctly. What if `s` is from a different stage than `b`? It should return 0.0, but now it might return something else. I think the easiest way to fix it is by creating a different belief type (perhaps called `StagedDiscreteBelief`) that stores the stage and checks it when `pdf` is called.